x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plan/logical/join/JoinTypes.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.esql.plan.logical.join;
+
+import org.elasticsearch.xpack.ql.expression.Attribute;
+
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Utility class defining the concrete types of joins supported by ESQL.
+ */
+public class JoinTypes {
+
+    private JoinTypes() {}
+
+    public static CoreJoinType INNER = CoreJoinType.INNER;
+    public static CoreJoinType LEFT = CoreJoinType.LEFT;
+    public static CoreJoinType RIGHT = CoreJoinType.RIGHT;
+    public static CoreJoinType FULL = CoreJoinType.FULL;
+    public static CoreJoinType CROSS = CoreJoinType.CROSS;
+
+    /**
+     * The predefined core join types. Implements as enum for easy comparison and serialization.
+     */
+    public enum CoreJoinType implements JoinType {
+        INNER("INNER"),
+        LEFT("LEFT OUTER"),
+        RIGHT("RIGHT OUTER"),
+        FULL("FULL OUTER"),
+        CROSS("CROSS");
```

Comment:
I'm a little uncomfortable adding all of these enum constants when we don't have any way to run them. But, like, these are the kinds of joins in SQL and they'd been like that for a while. So it's probably fine.